src/minecraftServers/Repositories/ServersReposirory.cs

```diff
@@ -1,29 +1,25 @@
 ﻿using MinecraftServers.Models;
+using MinecraftServers.Data;
+using Microsoft.EntityFrameworkCore;
 
 namespace MinecraftServers.Repositories;
 
 public class ServersReposirory : IServersRepository
 {
-    private readonly List<Server> _servers = new();
+    private readonly ServerContext _db;
+
+    public ServersReposirory(ServerContext context)
+    {
+        _db = context;
+    }
     public Server Add(Server server)
     {
         Random random = new Random();
         int serverOnline = random.Next(0, 999);
-        int numberOfServers = _servers.Count;
-        server.Id = Convert.ToUInt32(numberOfServers); 
-        if (server.Id == 0)
-        {
-            uint id = 0;
-            var newServer = new Server(server.Ip, serverOnline, server.Name, id);
-            if (_servers.Count == 0)
-            {
-                _servers.Add(newServer);
-                return newServer;
-            }
-        }
-        var newId = _servers.Last().Id + 1;
-        var nextServer = new Server(server.Ip, serverOnline, server.Name, newId);
-        _servers.Add(nextServer);
+        int numberOfServers = _db.Servers.Count();
+        var nextServer = new Server(server.Ip, serverOnline, server.Name, server.Id);
+        _db.Add(nextServer);
```

Comment:
You already have passed a server entity into a method, it's unnecessary to create object with same parameters